retrieval/transform.go

```diff
@@ -116,7 +116,7 @@ func (b *sampleBuilder) next(ctx context.Context, samples []tsdb.RefSample) (*mo
 		point.Value = &monitoring_pb.TypedValue{
 			Value: &monitoring_pb.TypedValue_DistributionValue{v},
 		}
```

Comment:
Should this be calling `updateSampleInterval`? It is called for every other metric type. This metric type is handled differently than the others, because it returns a successful status from the switch block, while the others only return in case of error. That seems to have led to the bug you fixed, which makes me suspect the lack of call to `updateSampleInterval` is another bug waiting to trigger.

Reply:
Good Catch. With the potential issue you pointed out, I updated the code to make the logic correct for `updateSampleInterval`, and refactor the variable to make the code more readable.